backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoicecandidate/api/impl/InvoiceCandDAO.java

```diff
@@ -573,6 +572,30 @@ public boolean isAvoidRecreate(final I_C_Invoice_Candidate ic)
 				.addColumn(I_C_Invoice_Candidate.COLUMN_C_Invoice_Candidate_ID);
 
 		return icQueryBuilder
+				.create()
+				.list();//
+	}
+
+	@NonNull
+	public List<I_C_Invoice_Candidate> retrieveInvoiceCandidates(@NonNull final InvoiceId invoiceId)
+	{
+		final I_C_Invoice invoice = load(invoiceId, I_C_Invoice.class);
+
+		final IQuery<I_C_Invoice_Line_Alloc> invoiceLineAllocQuery =
+				queryBL.createQueryBuilder(I_C_InvoiceLine.class, invoice)
+						.addEqualsFilter(I_C_InvoiceLine.COLUMNNAME_C_Invoice_ID, invoice.getC_Invoice_ID())
```

Comment:
```suggestion
		final IQuery<I_C_Invoice_Line_Alloc> invoiceLineAllocQuery =
				queryBL.createQueryBuilder(I_C_InvoiceLine.class, invoice)
						.addEqualsFilter(I_C_InvoiceLine.COLUMNNAME_C_Invoice_ID, invoiceId)
```

* the equals filter can handle `RepoIdAware`s
* we don't need to pass the invoice a context provider; the query-builder can get the thread-local current transaction or create a "local" one on the fly

Reply:
I'll drop the loading of the invoice then, thx!